File: app/src/test/java/com/example/adiphong/rx2/fromArrayTest.java
package com.example.adiphong.rx2;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Flowable;
import io.reactivex.Maybe;
import io.reactivex.Observable;
import io.reactivex.Single;

/**
 * Created by adiphong on 27/11/2017 AD.
 */

public class fromArrayTest {

    String[] testArray = {"hello", "world"};

    @Test
    public void flowableJust() {
        Flowable.fromArray(testArray)
                .subscribe(
                        s -> System.out.println("---------------flowable  result " + s)
                );
    }

    @Test
    public void observableJust() {
        Observable.fromArray(testArray)
                .subscribe(
                        s -> System.out.println("---------------observable result " + s)
                );

    }
}

File: app/src/test/java/com/example/adiphong/rx2/FromArrayTest.kt
package com.example.adiphong.rx2

import io.reactivex.Flowable
import io.reactivex.Observable
import org.junit.Test

/**
 * Created by adiphong on 27/11/2017 AD.
 */

class FromArrayTest {

    private var testArray = arrayOf("hello", "world")

    @Test
    fun flowableJust() {
        Flowable.fromArray(*testArray)
                .subscribe { s -> println("---------------flowable  result $s") }
    }

    @Test
    fun observableJust() {
        Observable.fromArray(*testArray)
                .subscribe { s -> println("---------------observable result $s") }
    }
}
